src/Graphs/partitionedgraphs/partitionedge.jl

```diff
@@ -0,0 +1,8 @@
+struct PartitionEdge{V,E<:AbstractEdge{V}} <: AbstractPartitionEdge{V}
+  edge::E
+end
+
+parent(pe::PartitionEdge) = getfield(pe, :edge)
+src(pe::PartitionEdge) = PartitionVertex(src(parent(pe)))
+dst(pe::PartitionEdge) = PartitionVertex(dst(parent(pe)))
+PartitionEdge(p::Pair) = PartitionEdge(NamedEdge(p.first => p.second))
```

Comment:
```suggestion
PartitionEdge(p::Pair) = PartitionEdge(NamedEdge(first(p) => last(p)))
```
I generally try to avoid accessing fields directly.